packages/peregrine/lib/targets/__tests__/peregrine-targets.spec.js

```diff
@@ -0,0 +1,83 @@
+const {
+    buildModuleWith,
+    mockTargetProvider
+} = require('@magento/pwa-buildpack');
+const declare = require('../peregrine-declare');
+const intercept = require('../peregrine-intercept');
+
+test('declares a sync target talons and intercepts transformModules', () => {
+    const targets = mockTargetProvider(
+        '@magento/peregrine',
+        (_, dep) =>
+            ({
+                '@magento/pwa-buildpack': {
+                    specialFeatures: {
+                        tap: jest.fn()
+                    },
+                    transformModules: {
+                        tap: jest.fn()
+                    }
+                }
+            }[dep])
+    );
+    declare(targets);
+    expect(targets.own.talons.tap).toBeDefined();
+    const hook = jest.fn();
+    // no implementation testing in declare phase
+    targets.own.talons.tap('test', hook);
+    targets.own.talons.call('woah');
+    expect(hook).toHaveBeenCalledWith('woah');
+
+    intercept(targets);
+    const buildpackTargets = targets.of('@magento/pwa-buildpack');
+    expect(buildpackTargets.transformModules.tap).toHaveBeenCalled();
+});
+
+test('enables third parties to wrap talons', async () => {
+    // sorry, buildModuleWith is slow. TODO: make it take less than a minute?
+    jest.setTimeout(60000);
```

Comment:
As we discussed this worries me, how common is it for developers to run all tests during their development cycle? Is there a way we could make some of these tests only run on PR to save developers the minute for this single test?

Reply:
We could! It's just not something we have set up currently. We could create a two-tiered test suite for Jest unit tests versus integration tests.

In normal workflow, a developer should be using `jest --watch`, which tests only files which have changed in the current Git repo state. That's been really slow in Jest for a while, though, so it's sometimes an unpopular choice.

Jest 26 really speeds it up, so I think that the full test suite might get a lot less common.